pkg/api/dryrunclient.go

```diff
@@ -87,12 +87,40 @@ func NewDryRunClient(apiClient client.APIClient, cli *command.DockerCli) (*DryRu
 // All methods and functions which need to be overridden for dry run.
 
 func (d *DryRunClient) ContainerAttach(ctx context.Context, container string, options moby.ContainerAttachOptions) (moby.HijackedResponse, error) {
-	return moby.HijackedResponse{}, ErrNotImplemented
+	return moby.HijackedResponse{}, errors.New("interactive run is not supported in dry-run mode")
 }
 
 func (d *DryRunClient) ContainerCreate(ctx context.Context, config *containerType.Config, hostConfig *containerType.HostConfig,
 	networkingConfig *network.NetworkingConfig, platform *specs.Platform, containerName string) (containerType.CreateResponse, error) {
-	return containerType.CreateResponse{}, ErrNotImplemented
+	projectLabel := fmt.Sprintf("com.docker.compose.project=%s", config.Labels["com.docker.compose.project"])
```

Comment:
better use const from https://github.com/docker/compose/blob/v2/pkg/api/labels.go

Reply:
I did the implementation of this PR in 2 phases, first the ContainerCreate changes and then I add to tweak the ContainerInspect function. In fact the changes in the 2nd one fix issues I faced with no existing containers. 
I'll simplify the the `ContainerCreate` implementation, it would now fake the creation of a new container